### DISCUSSION TITLE
meow + rectangle-mark-mode?

### DISCUSSION THREAD

**Discussion by gf3** [meow-edit/meow#263] [Q&A]
Does anyone have any recommendations on working with `rectangle-mark-mode`?

When marking a rectangle the standard meow functions don't seem to operate within the rectangle.

E.g. say I have the following snippet:

```go
	rules := govalidator.MapData{
		"code":                     []string{"required", "alpha_dash"},
		"disabled":                 []string{"required", "bool"},
		"expiresAt":                []string{"required"},
		"maxAmountToUse":           []string{"required", "numeric_between:0,"},
		"maxAmountToUsePerAccount": []string{"required", "numeric_between:0,"},
		"maxTimesToUse":            []string{"required", "numeric_between:0,"},
		"maxTimesToUsePerAccount":  []string{"required", "numeric_between:0,"},
		"startsAt":                 []string{"required"},
		"type":                     []string{"required"},
	}
```

And I'd like to change the column of strings with `"required"` I would typically mark a rectangle from the start of the first `"required"` until the end of the last `"required"` and then delete and insert. However `meow-change` will change the full lines between the start/end of the marked rectangle and `meow-delete` only operates on the singular character under the cursor.

**Comment by ml729** [comment]
I do not have an answer for your question regarding rectangle-mark-mode, but just wanted to say your edit can be done with meow's beacon state.
1. Select all the lines containing "required" (e.g. with `meow-line`)
2. Use `meow-grab`. (this activates Beacon state)
3. Use `meow-visit` to search for "required". (alternatively, navigate to the "required" string and press `meow-mark-word`)
4. Use `meow-change`; after returning to normal state your change should be applied to all the "required"s.

**Reply by gf3** [reply]
@ml729 thanks for the reply! i tried to use `meow-visit` while in Beacon state and it exited Beacon state and jumped me to a match further down the file. it would be very cool if `meow-visit` only worked within the selection—that would be very convenient

**Comment by DogLooksGood** [comment]
Well, you can use `meow-save`/`meow-kill` to copy/cut with `rectangle-mark-mode`. But meow dosen't use `rectangle-mark-mode` to edit, and `rectangle-mark-mode` itself is not about multiple cursors. 

https://user-images.githubusercontent.com/11796018/173566722-39265893-39b4-4a4c-b7e5-021fb33f3599.mp4
